resources/azure/compute/vm.go

```diff
@@ -74,23 +86,60 @@ func NewWindowsInstance(e azure.Environment, name, imageUrn, instanceType string
 		}
 	}
 
-	vm, nw, err := newVMInstance(e, name, imageUrn, instanceType, windowsOsProfile)
+	vm, nw, err := newVMInstance(e, name, imageUrn, instanceType, windowsOsProfile, opts...)
 	if err != nil {
-		return nil, nil, pulumi.StringOutput{}, err
+		return nil, pulumi.StringOutput{}, pulumi.StringOutput{}, err
 	}
 
-	return vm, nw, windowsAdminPassword.Result, nil
+	publicKey, err := os.ReadFile(e.DefaultPublicKeyPath())
+	if err != nil {
+		return nil, pulumi.StringOutput{}, pulumi.StringOutput{}, err
+	}
+
+	setupSSHCommand, err := compute.NewVirtualMachineRunCommandByVirtualMachine(e.Ctx(), fmt.Sprintf("%s-init-cmd", name), &compute.VirtualMachineRunCommandByVirtualMachineArgs{
+		ResourceGroupName: pulumi.String(e.DefaultResourceGroup()),
+		VmName:            vm.Name,
+		AsyncExecution:    pulumi.Bool(false),
+		RunCommandName:    pulumi.String("InitVM"),
+		Source: compute.VirtualMachineRunCommandScriptSourceArgs{
+			Script: pulumi.String(setupSSHScriptContent),
+		},
+		Parameters: compute.RunCommandInputParameterArray{
+			compute.RunCommandInputParameterArgs{
+				Name:  pulumi.String("authorizedKey"),
+				Value: pulumi.String(publicKey),
+			},
+		},
+		TimeoutInSeconds:                pulumi.Int(120),
+		TreatFailureAsDeploymentFailure: pulumi.Bool(true),
+	}, pulumi.Parent(vm))
+
+	if err != nil {
+		return nil, pulumi.StringOutput{}, pulumi.StringOutput{}, err
+	}
+
+	privateIP = pulumi.All(nw.IpConfigurations.Index(pulumi.Int(0)).PrivateIPAddress().Elem(), setupSSHCommand.URN()).ApplyT(func(args []interface{}) string {
+		return args[0].(string)
+	}).(pulumi.StringOutput)
```

Comment:
Note for reviewers: this is a hack to allow returning the private IP after the script is done. Otherwise we start trying to connect to the VM while it's not ready. Not a huge deal, feel free to remove and return directly the `privateIP`

Reply:
I think we could avoid that by making the `VM` resource a child of the `Network`. 
Like that we'd have
Connection waits for Network -> VM -> SSHSetup command
instead of 
Connection waits for Network -/-> VM -> SSH setup command
That can definitely be done in a follow up PR if this is working like that